machine/v1/types_controlplanemachineset.go

```diff
@@ -284,6 +290,24 @@ type GCPFailureDomain struct {
 	Zone string `json:"zone"`
 }
 
+// OpenStackFailureDomain configures failure domain information for the OpenStack platform.
+// +kubebuilder:validation:MinProperties:=1
+type OpenStackFailureDomain struct {
+	// availabilityZone is the nova availability zone in which the OpenStack machine provider will create the VM.
+	// If not specified, the VM will be created in the default availability zone specified in the nova configuration.
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
+	// rootVolume contains settings that will be used by the OpenStack machine provider to create the root volume attached to the VM.
+	// If not specified, no root volume will be created.
+	// If specified without a Zone, the root volume will be created in the availability specified by the volume type. If the volume type does not specify an
```

Comment:
`availability specified` -> `availability zone specified`

Reply:
> If the volume type does not specify an availability zone

Can you more elaborate on what it means to specify AZ in the volume type? E.g. "volumetype:availabilityzone" form?

Reply:
I added a few words and also a link to the doc. I hope this helps.